spec/unit/functions/logging_spec.rb

```diff
@@ -13,6 +13,14 @@ def collect_logs(code)
     Puppet::Util::Log.with_destination(Puppet::Test::LogCollector.new(logs)) do
       compiler.compile
     end
+
+    # There's a small chance that Facter could have generated some
+    # extra log messages during code compilation. This happened
+    # on Appveyor when the Appveyor image failed to resolve the
+    # domain fact. Since these messages are not relevant to our
+    # test, it is best to remove them from the collected logs.
+    logs.reject! { |log| log.source.downcase == "facter" }
+
```

Comment:
I'm thinking it would be better to change the tests to assert that the log collector contains **at least** the expected messages, as debug messages could be generated from sources other than Facter, and the expected message may not be first in the array. Something like:

```
expect(logs).to include(satisfy { |log| log.level == :notice && log.message == 'Undef' })
```

Or you could convert the log entries to hashes, but it's a little too magical and there's a symbol->string conversion:

```
expect(logs.map(&:to_hash)).to include(include('level' => 'notice', 'message' => 'Undef'))
```